src/gui/plotWidget.cpp

```diff
@@ -0,0 +1,242 @@
+    // make top and bottom axes be selected synchronously, and handle axis and tick labels as one
+    // selectable object:
+    if (customPlot->xAxis->selectedParts().testFlag(QCPAxis::spAxis) ||
+        customPlot->xAxis->selectedParts().testFlag(QCPAxis::spTickLabels) ||
+        customPlot->xAxis2->selectedParts().testFlag(QCPAxis::spAxis) ||
+        customPlot->xAxis2->selectedParts().testFlag(QCPAxis::spTickLabels)) {
+        customPlot->xAxis2->setSelectedParts(QCPAxis::spAxis | QCPAxis::spTickLabels);
+        customPlot->xAxis->setSelectedParts(QCPAxis::spAxis | QCPAxis::spTickLabels);
+    }
+    // make left and right axes be selected synchronously, and handle axis and tick labels as one
+    // selectable object:
+    if (customPlot->yAxis->selectedParts().testFlag(QCPAxis::spAxis) ||
+        customPlot->yAxis->selectedParts().testFlag(QCPAxis::spTickLabels) ||
+        customPlot->yAxis2->selectedParts().testFlag(QCPAxis::spAxis) ||
+        customPlot->yAxis2->selectedParts().testFlag(QCPAxis::spTickLabels)) {
+        customPlot->yAxis2->setSelectedParts(QCPAxis::spAxis | QCPAxis::spTickLabels);
+        customPlot->yAxis->setSelectedParts(QCPAxis::spAxis | QCPAxis::spTickLabels);
+    }
+}
+
+void Plot::beginNewGraph() {
+    customPlot->addGraph();
+}
+
+void Plot::mousePress() {
+    // if an axis is selected, only allow the direction of that axis to be dragged
+    // if no axis is selected, both directions may be dragged
+
+    if (customPlot->xAxis->selectedParts().testFlag(QCPAxis::spAxis))
+        customPlot->axisRect()->setRangeDrag(customPlot->xAxis->orientation());
+    else if (customPlot->yAxis->selectedParts().testFlag(QCPAxis::spAxis))
+        customPlot->axisRect()->setRangeDrag(customPlot->yAxis->orientation());
+    else
+        customPlot->axisRect()->setRangeDrag(Qt::Horizontal | Qt::Vertical);
+}
+
+void Plot::mouseWheel() {
+    // if an axis is selected, only allow the direction of that axis to be zoomed
+    // if no axis is selected, both directions may be zoomed
+
+    if (customPlot->xAxis->selectedParts().testFlag(QCPAxis::spAxis))
+        customPlot->axisRect()->setRangeZoom(customPlot->xAxis->orientation());
+    else if (customPlot->yAxis->selectedParts().testFlag(QCPAxis::spAxis)) {
+        autoRangeAction->setChecked(false);
+        customPlot->axisRect()->setRangeZoom(customPlot->yAxis->orientation());
+    } else {
+        autoRangeAction->setChecked(false);
+        customPlot->axisRect()->setRangeZoom(Qt::Horizontal | Qt::Vertical);
+    }
+}
+
+void Plot::mouseMove() {
+    if (QApplication::mouseButtons() & Qt::MouseButton::LeftButton) {
+        auto rangeDrag = customPlot->axisRect()->rangeDrag();
+        if (rangeDrag & Qt::Horizontal) {
+            autoShowNewestAction->setChecked(false);
+        }
+        if (rangeDrag & Qt::Vertical) {
+            autoRangeAction->setChecked(false);
+        }        
+    }
+}
+
+void Plot::addData(double x, double y) {
```

Comment:
Change signature to `(double time, double force)` as in `plotWidget.h`

Sync type with `void MainWindow::receiveNewForce(float time, float value)`; e.g. change every force variable to `double` as in Parser::Sample

https://github.com/linescaleGUI/linescaleGUI/blob/0afe4e680c2a405f207c1584da89e406e20217b8/src/parser/parser.h#L64-L72

Reply:
Will be updated separately in `feature/integrateParser`